### ISSUE TITLE
Cannot copy text in floating pane

### ISSUE THREAD

**Issue by Doctor-Scott** [omerxx/tmux-floax#5]
## Terminal
Wezterm
## shell
zsh
## Issue
It seems that no copy command is working within the tmux floating pane.
I have tried via the terminal, with `command-c` and tmux-vim keybinds

When I navigate to the scratch session in a normal tmux window it will allow me

I wonder if something in my existing config is clashing. 
I have included it below
Let me know if any other info is required

## tmux config
```
set-option -sa terminal-overrides ",xterm*:Tc"
set -g mouse on
unbind C-b
set -g prefix `
bind ` send-prefix


# Vim style pane selection

bind h resize-pane -L 10
bind j resize-pane -D 10
bind k resize-pane -U 10
bind l resize-pane -R 10

# Start windows and panes at 1, not 0
set -g base-index 1
set -g pane-base-index 1
set-window-option -g pane-base-index 1
set-option -g renumber-windows on

bind Left select-pane -L
bind Right select-pane -D
bind Up select-pane -U
bind Down select-pane -R

# Shift arrow to switch windows
bind -n S-Left  previous-window
bind -n S-Right next-window

# Shift Alt vim keys to switch windows
bind -n M-H previous-window
bind -n M-L next-window

set -g @catppuccin_flavour 'mocha'

set -g @plugin 'tmux-plugins/tpm'
set -g @plugin 'tmux-plugins/tmux-sensible'
set -g @plugin 'christoomey/vim-tmux-navigator'
set -g @plugin 'dreamsofcode-io/catppuccin-tmux'
set -g @plugin 'tmux-plugins/tmux-yank'
set -g @plugin 'tmux-plugins/tmux-open'

#Working with nvim zen mode
set-option -g allow-passthrough on

# tmux resurrect and continuum
set -g @plugin 'tmux-plugins/tmux-resurrect'
set -g @plugin 'tmux-plugins/tmux-continuum'
set -g @resurrect-strategy-nvim 'session'
# interval in mins
set -g @continuum-save-interval '1'
set -g @continuum-restore 'on'


#session x stuff
set -g @plugin 'omerxx/tmux-sessionx'
set -g @sessionx-bind 'o'
set -g @sessionx-zoxide-mode 'on'

set -g @sessionx-x-path '~/dotfiles/'

#floax stuff
set -g @plugin 'omerxx/tmux-floax'
set -g @floax-change-path 'false'
set -g @floax-bind 'BSpace'

# running this here so it doesnt override the below keymaps
run '~/.config/tmux/plugins/tpm/tpm'


# set vi-mode
set-window-option -g mode-keys vi
# keybindings
bind-key -T copy-mode-vi v send-keys -X begin-selection
bind-key -T copy-mode-vi C-v send-keys -X rectangle-toggle

# stop tmux leaving the scrollback buffer when highlighting with mouse or copying 
bind-key -T copy-mode-vi y send-keys -X copy-selection -x
unbind-key -T copy-mode-vi MouseDragEnd1Pane

bind C-k send-keys -R \; send-keys C-l \; clear-history

bind '"' split-window -v -c "#{pane_current_path}"
bind % split-window -h -c "#{pane_current_path}"

```

**Comment by feekApp** [comment]
I had the same issues using `tmux popup`. Seems like popup does not support clipboard.

**Comment by znd4** [comment]
hmm, I'm able to copy in popup windows. I think [these](https://github.com/znd4/xdg-config/blob/642eafd70acb88ef1a12f9330be8c684814970a2/tmux/tmux.conf?plain=1#L23-L25) are the relevant lines in my config. not sure though

**Comment by beatzball** [comment]
+1 - for me highlighting works inside the floating window, but triggering action to copy (in my case `Enter`) is not placing content in clipboard

**Comment by TimothyBesada** [comment]
I have the same issue.

**Comment by mohnasir-gh** [comment]
Facing a similar issue, not able to copy in the  terminal when inside tmux session

**Comment by Doctor-Scott** [comment]
So after some playing around, I found that this line in my config is causing the issue for me

```shell
# stop tmux leaving the scrollback buffer when copying 
bind-key -T copy-mode-vi y send-keys -X copy-selection -x
```

So I am leaving this commented out for now
I will deal with losing the position in the floating pane while I dig further into a workaround

**Comment by wojciech-kulik** [comment]
I'm facing the same issue. I select the text, and it should automatically copy it, but it doesn't happen within Floax. When I switch directly to that session outside of Floax, it works correctly.

**Comment by wojciech-kulik** [comment]
Any chance to fix that?

**Comment by wojciech-kulik** [comment]
It doesn't help in my case :(

**Comment by feekApp** [comment]
still expect the issue is related to TMUX popup. In all my popups there is no clipboard available

**Comment by Doctor-Scott** [comment]
> It doesn't help in my case :(

Maybe start with a empty tmux config with only what is needed for the floax plugin?

See if you can copy text with that setup

If so, add in more plugins/ config until it stops, that's mostly likely the specific cause for yourself

**Comment by wojciech-kulik** [comment]
My config is very simple. However, I removed everything, just left resurrect, continuum & floax, then I killed tmux server and opened it again. Still doesn't work.

> that's mostly likely the specific cause for yourself

How about all other people here :)?

**Comment by gambhirsharma** [comment]
Any solution to this issue? 

I’m currently using `pbcopy` on my Mac as a workaround, but I’m looking for a permanent solution.

**Comment by wojciech-kulik** [comment]
Could you tell what’s your workaround? You just call echo and redirect it to pbcopy?

**Comment by gambhirsharma** [comment]
It's simple just use the `|` pipe operator with `pbcopy` to pass the output to it. Eg: `pnpm run dev | pbcopy`. Or sometimes `pnpm run dev > log.txt` to save it in a file. 

But a major limitation of this approach is that it doesn’t capture standard error. But there is a workaround `pnpm run dev 2>&1 | pbcopy` to redirect from standard error `2` to standard output `&1`

**Comment by wojciech-kulik** [comment]
Yep, it works well with the output, but most of the time I want to copy the selection. That's the biggest issue.

**Comment by wojciech-kulik** [comment]
@raeperd thank you so much, my misery finally has ended ♥️

Based on your solution, here is the fix for copy selection using mouse drag:

```
bind-key -T copy-mode-vi MouseDragEnd1Pane send -X copy-pipe-and-cancel "pbcopy"
```

**Comment by robotnix5** [comment]
> I had the same issues using `tmux popup`. Seems like popup does not support clipboard.

I investigated this pretty closely. I isolated the issue to [this command](https://github.com/omerxx/tmux-floax/blob/main/scripts/utils.sh#L104)
```bash
    tmux popup \
        -S fg="$FLOAX_BORDER_COLOR" \
        -s fg="$FLOAX_TEXT_COLOR" \
        -T "$FLOAX_TITLE" \
        -w "$FLOAX_WIDTH" \
        -h "$FLOAX_HEIGHT" \
        -b rounded \
        -E \
        "tmux attach-session -t \"$FLOAX_SESSION_NAME\"" 
``` 

When I run the command `tmux popup -E "tmux attach"`, the expected session is attached in the popup window. While in the popup window, I enter copy mode, select some text, and hit <enter>. I can see the selected text if I run `:showb`, but the text is not in my system clipboard. 
If I run `tmux popup`, I can repeat the above steps to copy text and the text is in my system clipboard.
If I run `tmux attach` with no sessions attached, it successfully restores my tmux sessions and copies text to the clipboard.

This [solution](https://github.com/omerxx/tmux-floax/issues/5#issuecomment-2581884442) from @raeperd solved my drag-to-copy issue. However, I still could not send text to the clipboard from my text editor. I emailed the tmux google group for answers. This is what they told me

> Clipboard escape sequences (and a few others) are not currently supported in popups.

So at it's core, this is a tmux limitation
